maturin strips `PYO3_CROSS_LIB_DIR` from the build command env

### Bug Description

When cross-compiling a Windows wheel by calling `cargo build` everything works correctly:
```
PYO3_CROSS_LIB_DIR=/x/python3-dll cargo build --release --target x86_64-pc-windows-gnu
```

When using `maturin build` as follows:
```
PYO3_CROSS_LIB_DIR=/x/python3-dll maturin build --release --strip --target x86_64-pc-windows-gnu
```

I get the following error:
```
  = note: /usr/lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld: cannot find -lpython3
          collect2: error: ld returned 1 exit status

💥 maturin failed
  Caused by: Failed to build a native library through cargo
  Caused by: Cargo build finished with "exit status: 101": `cargo rustc --message-format json --manifest-path Cargo.toml --target x86_64-pc-windows-gnu --release --lib -- -C link-arg=-s -L native=/x/python3-dll`
```

The error is occurring when linking one of the dependencies, not the target crate itself.
This dependency is both a PyO3 extension module and a Rust library, and cargo always generates both outputs.
Of these, only the Rust library is used to link the target extension module.

If I re-add the missing `PYO3_CROSS_LIB_DIR` variable the failing command succeeds:
```
PYO3_CROSS_LIB_DIR="$PWD"/target/python3-dll cargo rustc --message-format json --manifest-path Cargo.toml --target x86_64-pc-windows-gnu --release --lib -- -C link-arg=-s -L native=/home/kvachenok_sv/pyspock-legacy/target/python3-dll
```

Maybe maturin should just pass `PYO3_CROSS_LIB_DIR`  to `cargo rustc` directly and not try to modify the library search path manually here:

https://github.com/PyO3/maturin/blob/8c849a19e7a848e6e024673baffdb6f26663b4f3/src/compile.rs#L179-L194

### Your Python version (`python -V`)

Python 3.10.0

### Your pip version (`pip -V`)

pip 21.0

### What bindings you're using

pyo3

### Does `cargo build` work?

- [X] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [X] Yes

### Steps to Reproduce

1. Try to cross-compile Windows a PyO3 extension crate that also has a PyO3 extension crate as a dependency.
2. The dependency fails to build.

--- a/src/compile.rs
+++ b/src/compile.rs
@@ -237,7 +237,7 @@ fn compile_target(
         let is_pypy = python_interpreter
             .map(|p| p.interpreter_kind == InterpreterKind::PyPy)
             .unwrap_or(false);
-        if !is_pypy {
+        if !is_pypy && !target.is_windows() {
             // This will make pyo3's build script only set some predefined linker
             // arguments without trying to read any python configuration
             build_command.env("PYO3_NO_PYTHON", "1");

